doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,194 @@
+# Maintaining the Build files
+
+This document explains how to maintain the build files in the codebase.
+
+## Overview
+
+On how to build the Node.js core, see [Building Node.js](../../BUILDING.md).
+
+There are three main build files that will be directly run to build Node.js:
+
+On Windows:
+
+- `vcbuild.bat`: A Windows Batch Script that locates build tools, provides
+  a subset of the targets avilable in the [Makefile](#makefile), and a few
+  targets of its own. This script also calls [configure](#configure) before
+  building the Node.js binary. For a detailed guide on this script,
+  see [vcbuild.bat](#vcbuild.bat).
+
+On unix-like systems:
+
+- `configure`: A (mostly) Python 2 script that accepts a set of options,
```

Comment:
I'd move this to it's own sub-section, since _it is_ run on Windows (it's just mostly indirectly).
```markdown
`configure`: A Python script that detects system capabilities and runs [GYP][]. The scripts generate `config.gypi` which includes parameters used by GYP to create system dependant build files. Output is usually in one this formats: Makefile, MSbuild, ninja, or XCode. For a detailed guide on this script, see [configure](#configure).


[GYP]: https://gyp.gsrc.io/docs/UserDocumentation.md
```